plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/ObjectWriteFunction.java

```diff
@@ -27,7 +27,7 @@
     void set(PreparedStatement statement, int index, Object value)
             throws SQLException;
 
-    static <T> ObjectWriteFunction of(Class<T> javaType, ObjectWriteFunctionImplementation<T> implementation)
+    static <T> ObjectWriteFunction of(Class<T> javaType, int jdbcType, ObjectWriteFunctionImplementation<T> implementation)
```

Comment:
Revert. In The two places where you actually use it, it's fine to implement `ObjectWriteFunction` interface directly.